### PR TITLE
Bugfix update channels

### PR DESCRIPTION
**Pull request by camilasan**
I was just hoping to catch the issue of reports about it - like the channel changing without the user interaction.

### CODE REVIEW

File: src/gui/generalsettings.cpp [hunk 1]
```diff
@@ -284,18 +288,21 @@ void GeneralSettings::loadMiscSettings()
...
-    connect(_ui->updateChannel, &QComboBox::currentTextChanged, this, &GeneralSettings::slotUpdateChannelChanged);
-#endif
+void GeneralSettings::loadUpdateChannelsList() {
+    ConfigFile cfgFile;
+    if (_currentUpdateChannelList != cfgFile.validUpdateChannels()) {
+        _currentUpdateChannelList = cfgFile.validUpdateChannels();
+        _ui->updateChannel->clear();
+        _ui->updateChannel->addItems(_currentUpdateChannelList);
+        const auto currentUpdateChannelIndex = _currentUpdateChannelList.indexOf(cfgFile.currentUpdateChannel());
+        _ui->updateChannel->setCurrentIndex(currentUpdateChannelIndex != -1? currentUpdateChannelIndex : 0);
```
Comment (claucambra) [inline src/gui/generalsettings.cpp:301]: nit: spacing

```suggestion
        _ui->updateChannel->setCurrentIndex(currentUpdateChannelIndex != -1 ? currentUpdateChannelIndex : 0);
```

File: src/gui/settingsdialog.cpp [hunk 2]
```diff
@@ -137,8 +139,8 @@ SettingsDialog::SettingsDialog(ownCloudGui *gui, QWidget *parent)
     _actionGroupWidgets.insert(generalAction, generalSettings);
     _actionGroupWidgets.insert(networkAction, networkSettings);
 
-    foreach(auto ai, AccountManager::instance()->accounts()) {
-        accountAdded(ai.data());
+    foreach(auto account, AccountManager::instance()->accounts()) {
```
Comment (claucambra) [inline src/gui/settingsdialog.cpp:142]: Better to use a ranged for loop here

```suggestion
    for(const auto account : std::as_const(AccountManager::instance()->accounts())) {
```

**Review by claucambra — COMMENTED** [review]
Some nitpicks otherwise looks good imo

**Review by mgallien — CHANGES_REQUESTED** [review]
when built with `BUILD_UPDATER=OFF`, build failed as expected from my inline comment

**Review by mgallien — APPROVED** [review]

**Pull request merged**